### PR TITLE
fix: Expose missing Add/RemoveExtraParameter methods to macOS node child processes

### PR DESCRIPTION
**Pull request by lepinay**
##### Description of Change
This will expose the missing methods on crashReporter when running in a child node process (fork/spawn) on mac:
AddExtraParameter
RemoveExtraParameter

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

##### Release Notes
Notes: Fixed crashReporter `addExtraParameter` / `removeExtraParameter` methods `undefined` in macOS node child processes

### CODE REVIEW

File: spec/api-crash-reporter-spec.js [hunk 1]
```diff
@@ -166,11 +167,49 @@ describe('crashReporter module', () => {
         })
       })
 
+      it('should send minidump with updated extra parameters when node processes crash', function (done) {
```
Comment (miniak) [inline spec/api-crash-reporter-spec.js]: when node processes crashes

File: spec/fixtures/api/crash.html [hunk 2]
```diff
@@ -20,7 +20,7 @@
     })
 
     if (process.platform === 'win32') {
-      ipcRenderer.sendSync('crash-service-pid', crashReporter.crashServicePid)
+      ipcRenderer.sendSync('crash-service-pid', crashReporter._crashServiceProcess.pid)
```
Comment (miniak) [inline spec/fixtures/api/crash.html]: why are you changing this?

Comment (miniak) [reply spec/fixtures/api/crash.html]: `_crashServiceProcess` is not defined

Comment (miniak) [reply spec/fixtures/api/crash.html]: https://github.com/electron/electron/commit/508551b81b3e76998b38e6d56aaa0de47e488664#diff-ba522aa2d4f3759eca6524511f8b9442

File: spec/fixtures/api/crash.html [hunk 3]
```diff
@@ -20,7 +20,7 @@
     })
 
     if (process.platform === 'win32') {
-      ipcRenderer.sendSync('crash-service-pid', crashReporter.crashServicePid)
```
Comment (miniak) [inline spec/fixtures/api/crash.html]: please revert this change, `crashServicePid` was fixed in https://github.com/electron/electron/pull/15206

**Comment by lepinay** [comment]
Hi @miniak , just bumping this thread, change was reverted based on review, let me know if anything else needed.
Cheers

File: atom/app/node_main.cc [hunk 4]
```diff
@@ -5,6 +5,7 @@
 #include "atom/app/node_main.h"
 
 #include <memory>
+#include <string>
```
Comment (shahzadlone) [inline atom/app/node_main.cc]: Perhaps if strings only needed on MAC OS, might want to put the include in the ifdef.

Comment (miniak) [reply atom/app/node_main.cc]: Not really necessary

**Comment by miniak** [comment]
@lepinay can you please rebase on top of latest master?

**Comment by lepinay** [comment]
Hi @miniak, rebase to master is done

**Comment by emmkimme** [comment]
Hi @miniak
this MR is on-hold since Nov. 2018, could it be possible to merge it please
Thanks

**Comment by miniak** [comment]
@emmkimme please rebase again? @codebytere can you help merging the PR?

**Comment by lepinay** [comment]
Hi, @miniak @codebytere rebase is done.

**Comment by miniak** [comment]
@lepinay needs rebasing again :(

**Comment by miniak** [comment]
@lepinay would it now also work on Windows after #18483

**Comment by miniak** [comment]
https://github.com/electron/electron/blob/master/docs/api/crash-reporter.md#crashreporteraddextraparameterkey-value-macos-windows
https://github.com/electron/electron/blob/master/docs/api/crash-reporter.md#crashreporterremoveextraparameterkey-macos-windows

**Comment by lepinay** [comment]
@miniak rebase is done
Add/RemoveExtraParameter is already working on windows for node child processes

**Comment by miniak** [comment]
@lepinay how does it work?

**Comment by lepinay** [comment]
@miniak 
My bad, I forgot I added an additional layer on my side that make it "works" on windows by restarting the crash reporter on windows ;)

So, in that case https://github.com/electron/electron/pull/18483 will need to add the support for it in the Windows implementation. It will make more sense to put it there. Because my PR is more a bug fix for the current OSX implementation of crashpad and could be fixed now indepently of https://github.com/electron/electron/pull/18483.

Or maybe we can create a separate PR once https://github.com/electron/electron/pull/18483 has been merged to add this support on windows as well  ?

**Comment by miniak** [comment]
I would modify this PR. It should be basically changing `#if defined(OS_MACOSX)` to `#if defined(OS_LINUX)`, right?

**Comment by lepinay** [comment]
@miniak Yes now that https://github.com/electron/electron/pull/18483 is merged, that would be the change to make indeed + testing that it works on windows.

**Comment by miniak** [comment]
@lepinay please rebase and do the change as requested

**Comment by lepinay** [comment]
@miniak done, please review.

**Comment by miniak** [comment]
@zcbenz let’s get this merged before 7 is branched
